src/peripheral/component/attach.clj

```diff
@@ -51,7 +51,7 @@
   ([component k attach-component]
    (attach component k attach-component nil))
   ([component k attach-component dependencies]
-   (assert (not (contains? component k)) "there is already a component with that key attached.")
+   (assert (not (contains? component k)) (str "there is already a component with " k " key attached."))
```

Comment:
Grammatically, I think it should be "there is already a component with key XY attached". (Currently it's "[...] with XY key [...]".)